src/main/java/duke/Save.java

```diff
@@ -0,0 +1,89 @@
+package duke;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.FileNotFoundException;
+import java.security.KeyStore;
+import java.util.Scanner;
+
+public class Save {
+    public static final int LENGTH_MARKER = 4;
+    public static String filePath = "data.txt";
+    public static String originalInputPath = "inputFile.txt";
+    public static TaskManager taskList = new TaskManager();
+    // load data to this taskList and pass the reference of the taskList to the one in Duke.java
+
+    public static void saveData(String dataToWrite, String path) {
+        try {
+            FileWriter fw = new FileWriter(path);
+            fw.write(dataToWrite);
+            fw.close();
+        } catch (IOException e) {
+            System.out.println(e.getMessage());
+        }
+    }
+
+    public static void loadData() throws IOException {
+        File f = new File(filePath);
+        File inputFile = new File(originalInputPath);
+        if(f.createNewFile()) {
+            System.out.println("Welcome! This is the first time for you to use Duke! Enjoy!");
+        } else {
+            try{
+                loadAllTasksList();   //add tasks and do the marking
+                System.out.println("Welcome bake to Duke, your task list saved was successfully loaded!");
+            } catch (FileNotFoundException e) {
+                System.out.println(e.getMessage());
```

Comment:
You have a try-catch block to handle `FileNotFound` exception, but the code doesn't show how the file is being used. This breaks the[ logical structure of the code. ](https://nus-cs2113-ay2122s1.github.io/website/se-book-adapted/chapters/codeQuality.html#structure-code-logically)

How about using the try-catch in `loadAllTasksList()` instead?